### PR TITLE
Add functionality for users to retrieve template by Name as well as UUID

### PR DESCRIPTION
**Pull request by mmlb**
## Description

This PR adds the ability for Tink to retrieve workflows by either UUID or by Name.
If the value after `tink template get` is not a valid UUID, Tink will default to searching for the value in the Name field of the templates.

Mostly developed by @Belchy06, thanks!
My additions are:
 - cleaned up the import blocks so its just 2 blocks
 - fix typo in help output
 - rebase on latest main
 - rework some of the logic in cmd/tink-cli/cmd/get/get.go to be clearer

## Why is this needed

This feature is required for a project @Belchy06 is working on but is also something I'd like to make use of.

## How Has This Been Tested?

This has been tested in the VM's based on the [Local Setup with Vagrant](https://docs.tinkerbell.org/setup/local-vagrant/) tutorial, as well as in a personal deployment by @Belchy06.

### CODE REVIEW

File: cmd/tink-cli/cmd/get/get.go [hunk 1]
```diff
@@ -155,3 +148,31 @@ func NewGetCommand(opt Options) *cobra.Command {
 	opt.clientConnOpt.SetFlags(cmd.PersistentFlags())
 	return cmd
 }
+
+func retrieveMulti(opt Options, cmd *cobra.Command, args []string) ([]interface{}, error) {
```
Comment (mmlb) [inline cmd/tink-cli/cmd/get/get.go:152]: I had to extract this into a separate function to get golangci-linter happy.

**Review by nshalman — COMMENTED** [review]
This all looks reasonable to me. Any idea why CI is failing, @mmlb?

**Comment by jacobweinstock** [comment]
Hey @mmlb, is this still on your radar? Any help needed?

**Comment by mmlb** [comment]
> Hey @mmlb, is this still on your radar? Any help needed?

Still on the radar not sure why I hadn't merged it yet. I think I wanted to give it a go via sandbox but forgot/got distracted with boot issues I was debugging :D. I'll get CI to pass and get it merged soon.

**Comment by mmlb** [comment]
> This all looks reasonable to me. Any idea why CI is failing, @mmlb?

Rebased on top of master and now failing on actions/setup-go not being nice and exporting `GOROOT` which messes with the go being run within nix-shell. I'm going to open up a PR to actions/setup-go so it stops doing this.

**Comment by mmlb** [comment]
Waiting on https://github.com/actions/setup-go/pull/175

**Comment by mmlb** [comment]
I'll rebase on top of main now that https://github.com/tinkerbell/tink/pull/582 should be unblocking this.

**Comment by jacobweinstock** [comment]
Hey @mmlb, just a ping on this one.

**Comment by mmlb** [comment]
Thanks for the ping @jacobweinstock, lets see how things go.

**Comment by mmlb** [comment]
ah the tests are still going to fail though, lets see if thats a quick fix.

**Comment by mmlb** [comment]
alright @jacobweinstock I'm all rebased and tests passing!

**Review by jacobweinstock — APPROVED** [review]

**Comment by mmlb** [comment]
Hey @jacobweinstock ptal again. I added a couple more tests, for when the function that fetches the data doesn't exist. And added some protection to avoid running an undefined function.

**Review by jacobweinstock — APPROVED** [review]

**Comment by Belchy06** [comment]
Awesome to see this merged in.

Sorry for going MIA, my studies and work absolutely swamped me and I was unable to set some time aside to get the original PR complete.

Thanks guys!

**Comment by mmlb** [comment]
@Belchy06 no worries, thanks for kicking this off.

**Pull request merged**